tests/mocks/mockAuthorizationService.js

```diff
@@ -0,0 +1,102 @@
+const express = require("express");
+const jose = require("node-jose");
+const uuid = require("uuid");
+
+const logPrefix = "[AIRLOCK DEV]";
+const keyStore = jose.JWK.createKeyStore();
+const PORT = 8001;
+
+async function server() {
+    const app = express();
+
+    app.use(express.json());
+
+    app.use("/oauth2/default/v1/token", async (req, res, next) => {
+        let jwt;
+        try {
+            jwt = await createJWT();
+        } catch (error) {
+            return res.status(500).send({
+                error: error?.message || "Undefined Error",
+                ts: Date.now()
+            });
+        }
+        return res.status(200).send({
+            "token": jwt
+        });
+    });
+
+    app.use("/oauth2/default/v1/keys", (req, res, next) => {
+        return res.status(200).send(
+            keyStore.toJSON()
+        );
+    });
+
+    app.listen(PORT, () => {
+        console.log(logPrefix, `Airlock Dev authorization service listening on port ${PORT}`);
+    });
+}
+
+async function createJWK() {
+    try {
+        await keyStore.generate("RSA", 2048, {
+            kid: uuid.v4(),
+            alg: "RS256",
+            use: "sig"
+        });
+    } catch (error) {
+        throw ({ message: error?.message || "Cannot generate a new JWK." });
```

Comment:
we should only `throw new Error(...)` and not objects, nor rethrow previous errors.
Throwing a new error will ensure that we capture the stack trace at the point where the error is thrown.